### ISSUE TITLE
BEST

### ISSUE THREAD

**Issue by cleong110** [cleong110/sign-language-processing.github.io#19]
https://ojs.aaai.org/index.php/AAAI/article/view/25470

**Comment by cleong110** [comment]
Official Citation: 

```
@article{Zhao_Hu_Zhou_Shi_Li_2023, title={BEST: BERT Pre-training for Sign Language Recognition with Coupling Tokenization}, volume={37}, url={https://ojs.aaai.org/index.php/AAAI/article/view/25470}, DOI={10.1609/aaai.v37i3.25470}, abstractNote={In this work, we are dedicated to leveraging the BERT pre-training success and modeling the domain-specific statistics to fertilize the sign language recognition~(SLR) model. Considering the dominance of hand and body in sign language expression, we organize them as pose triplet units and feed them into the Transformer backbone in a frame-wise manner. Pre-training is performed via reconstructing the masked triplet unit from the corrupted input sequence, which learns the hierarchical correlation context cues among internal and external triplet units. Notably, different from the highly semantic word token in BERT, the pose unit is a low-level signal originally locating in continuous space, which prevents the direct adoption of the BERT cross entropy objective. To this end, we bridge this semantic gap via coupling tokenization of the triplet unit. It adaptively extracts the discrete pseudo label from the pose triplet unit, which represents the semantic gesture / body state. After pre-training, we fine-tune the pre-trained encoder on the downstream SLR task, jointly with the newly added task-specific layer. Extensive experiments are conducted to validate the effectiveness of our proposed method, achieving new state-of-the-art performance on all four benchmarks with a notable gain.}, number={3}, journal={Proceedings of the AAAI Conference on Artificial Intelligence}, author={Zhao, Weichao and Hu, Hezhen and Zhou, Wengang and Shi, Jiaxin and Li, Houqiang}, year={2023}, month={Jun.}, pages={3597-3605} }

```

**Comment by cleong110** [comment]
Reformatted: 
```
@article{Zhao_Hu_Zhou_Shi_Li_2023,
  title        = {BEST: BERT Pre-training for Sign Language Recognition with Coupling Tokenization},
  volume       = {37},
  url          = {https://ojs.aaai.org/index.php/AAAI/article/view/25470},
  doi          = {10.1609/aaai.v37i3.25470},
  number       = {3},
  journal      = {Proceedings of the AAAI Conference on Artificial Intelligence},
  author       = {Zhao, Weichao and Hu, Hezhen and Zhou, Wengang and Shi, Jiaxin and Li, Houqiang},
  year         = {2023},
  month        = {Jun.},
  pages        = {3597-3605}
}
```

**Comment by cleong110** [comment]
Abstract: 
>In this work, we are dedicated to leveraging the BERT pre-training success and modeling the domain-specific statistics to fertilize the sign language recognition~(SLR) model. Considering the dominance of hand and body in sign language expression, we organize them as pose triplet units and feed them into the Transformer backbone in a frame-wise manner. Pre-training is performed via reconstructing the masked triplet unit from the corrupted input sequence, which learns the hierarchical correlation context cues among internal and external triplet units. Notably, different from the highly semantic word token in BERT, the pose unit is a low-level signal originally locating in continuous space, which prevents the direct adoption of the BERT cross entropy objective. To this end, we bridge this semantic gap via coupling tokenization of the triplet unit. It adaptively extracts the discrete pseudo label from the pose triplet unit, which represents the semantic gesture / body state. After pre-training, we fine-tune the pre-trained encoder on the downstream SLR task, jointly with the newly added task-specific layer. Extensive experiments are conducted to validate the effectiveness of our proposed method, achieving new state-of-the-art performance on all four benchmarks with a notable gain.

**Comment by cleong110** [comment]
![image](https://github.com/cleong110/sign-language-processing.github.io/assets/122366389/c663ac7e-18fd-4724-a21e-cc82774ec98b)

**Comment by cleong110** [comment]
c.f. #14

**Comment by cleong110** [comment]
Building a ChatGPT Prompt

```
I am writing a summary of an academic paper. Based on what I have provided below, can you rewrite my first version of the summary to be more concise and professional? Please provide 3 alternative rewrites, and explain your suggested changes, as well as any issues with writing quality or inaccuracy in my original summary. Be sure the summaries you provide are accurate to the figure and the the abstract. If I have missed a key contribution from the paper please note that and suggest additions. If something is not clear request clarification and I can provide additional snippets. Please cite your sources for important details, e.g. "from the abstract" or "based on the full text". My summary is in markdown syntax and contains citations to a BibTex bibliography, the citations begin with "@". Please use the same citation style.

In addition, please follow the following style guide:

STYLE GUIDE
- **Citations**: Use the format `@authorYearKeyword` for inline citations, and `[@authorYearKeyword]` for citations wrapped in parentheses. To include multiple citations ,use a semicolon (;) to separate them (e.g., "@authorYearKeyword;@authorYearKeyword").
- **Background & Related Work**: Use simple past tense to describe previous work (e.g., "@authorYearKeyword used...").
- **Abbreviations**: Define abbreviations in parentheses after the full term (e.g., Langue des Signes Française (LSF)).
- **Percentages**: Use the percent sign (%) with no space between the number and the sign (e.g., 95%).
- **Spacing**: Use a single space after periods and commas.
- **Hyphenation**: Use hyphens (-) for compound adjectives (e.g., video-to-pose).
- **Lists**: Use "-" for list items, followed by a space.
- **Code**: Use backticks (`) for inline code, and triple backticks (```) for code blocks.
- **Numbers**: Spell out numbers less than 10, and use numerals for 10 and greater.
- **Contractions**: Avoid contractions (e.g., use "do not" instead of "don't").
- **Compound Words**: Use a forward slash (/) to separate alternative compound words (e.g., 2D / 3D).
- **Phrasing**: Prefer active voice over passive voice (e.g., "The authors used..." instead of "The work was used by the authors...").
- **Structure**: Present information in a logical order.
- **Capitalization**: Capitalize the first word of a sentence, and proper nouns.
- **Emphasis**: Use italics for emphasis by wrapping a word with asterisks (e.g., *emphasis*).
- **Quote marks**: Use double quotes (").
- **Paragraphs**: When a subsection header starts with ######, add "{-}" to the end of the subsection title to indicate a new paragraph. If it starts with #, ##, ###, ####, or ##### do not add the "{-}".
- **Mathematics**: Use LaTeX math notation (e.g., $x^2$) wrapped in dollar signs ($).


All right, here is information about the paper I am trying to summarize:

Paper Title: ""

Abstract: 
""

Full Text: see attached PDF

My Summary:  
""

All right, remember my initial instructions, please go ahead and provide me the requested concise, professional rewrite suggestions for my summary, with the requested explanations, citations, and following the style guide.
```

**Comment by cleong110** [comment]
Back to BEST: 

>Our contributions are summarized as follows, • We propose a self-supervised pre-trainable framework. It leverages the BERT success, jointly with the specific design for the sign language domain. The Thirty-Seventh AAAI Conference on Artificial Intelligence (AAAI-23) 359

• We organize the main hand and body movement as the pose triplet unit and propose the masked unit modeling (MUM) pretext task. To utilize the BERT objective, we generate the pseudo label for this task via coupling tokenization on the pose triplet unit. • Extensive experiments on downstream SLR validate the effectiveness of our proposed method, achieving new state-of-the-art performance on four benchmarks with a notable gain.

**Comment by cleong110** [comment]
Datasets: 

>We conduct experiments on four public sign language datasets, i.e., NMFs-CSL (Hu et al. 2021b), SLR500 (Huang et al. 2018), WLASL (Li et al. 2020a) and MSASL (Joze and Koller 2018).

**Comment by cleong110** [comment]
![image](https://github.com/cleong110/sign-language-processing.github.io/assets/122366389/df061a51-468f-4344-a041-34054543bf71)

**Comment by cleong110** [comment]
![image](https://github.com/cleong110/sign-language-processing.github.io/assets/122366389/959d44cb-7b81-4c97-9f11-e9c649775325)
They compare with SignBERT (but not SignBERT+?)

**Comment by cleong110** [comment]
Honestly, I think this one should not be merged until #14 is merged

**Comment by cleong110** [comment]
Interesting: 
> Notably, different from the highly semantic word token in BERT, the pose unit is a low-level signal originally located in continuous space, which prevents the direct adoption of the BERT cross-entropy objective. To this end, we bridge this semantic gap via coupling tokenization of the triplet unit.

OK, what's that mean?

**Comment by cleong110** [comment]
> However, the main obstacle to leverage its success in video SLR is the different characteristics of the input signal. In NLP, the input word token is discrete and pre-defined with high semantics. In contrast, the video signal of sign language is continuous with the spatial and temporal dimensions. This signal is quite lowlevel, making the original BERT objective not applicable. Besides, since the sign language video is mainly characterized by hand and body movements, the direct adoption of the BERT framework may not be optimal.

**Comment by cleong110** [comment]
"Pose Triplet Unit"?

**Comment by cleong110** [comment]
Oh this seems important: 
>The tokenization provides pseudo labels for our designed pretext task during pre-training. We utilize a discrete variational autoencoder (d-VAE) to jointly convert the pose triplet unit into the triplet tokens (body, left and right hand), motivated by VQ-VAE (Van Den Oord, Vinyals et al. 2017).

**Comment by cleong110** [comment]
![image](https://github.com/cleong110/sign-language-processing.github.io/assets/122366389/0c39262b-c4ed-4076-b9a6-1f0ad64c5de9)

**Comment by cleong110** [comment]
![image](https://github.com/cleong110/sign-language-processing.github.io/assets/122366389/adf23dad-1617-4285-88db-f5559d6ef814)

**Comment by cleong110** [comment]
>Our designed pretext task is MUM, which aims to exploit the hierarchical correlation context among internal and external triplet pose units. Given a pose sequence with a triplet pose unit of length T , we first randomly choose the α · T frames to process the mask operation. For clarification, we define three parts of the pose triplet unit as f l sign,t, f r sign,t and fb sign,t, respectively. If a unit is masked, a learnable masked token emask ∈ RDpart is utilized to replace each part of the triplet unit with 50% probability. Therefore, the masked triplet unit includes three masking cases: only hand masked, only body masked and hand-body masked.

**Comment by cleong110** [comment]
OK, in my own words now, real informally: 

So the thing to know about BEST is that they wanted to do BERT-style masked language modeling, but, you know, BERT assumes you've already got discrete, sematically meaningful tokens. 
So they were, like, well we've got these triplets of left hand, right hand, body (no face keypoints, mind you!), let's make those into triplets and couple them together. 
OK so then what? Well, van den Oord and Vinyals wrote "Neural discrete representation learning", which lets you take continuous signals and make them into discrete codes, with like a codebook and stuff.
So they use one of things to make a tokenizer, they put the coupled triplets into it and get discrete tokens out. 
OK then what? Well then you've got the discrete tokens, and you mask hand, body, or both, and the transformer has to reconstruct the correct hand position, or body position, or whatever from surrounding context. 
"You shall know a ~word~ handshape from the company it keeps" I suppose.
(and you use positional encodings to inform the model about temporal stuff, SignBERT+ does that too)
And then they tried it on Isolated SLR and it seemed to work pretty good.

**Comment by cleong110** [comment]
Also let's do a quick compare/contrast...

SignBERT+ results: 
![image](https://github.com/cleong110/sign-language-processing.github.io/assets/122366389/461b49b8-93cb-4fd8-b325-208b898fd5b9)

BEST results: 
![image](https://github.com/cleong110/sign-language-processing.github.io/assets/122366389/2d7d9157-ca23-475d-af04-fcdc866cdedc)

WINNER: SignBERT+

**Comment by cleong110** [comment]
#### WLASL

SignBERT+
![image](https://github.com/cleong110/sign-language-processing.github.io/assets/122366389/2256ab6e-6f24-4ec6-801c-fd5866241196)

BEST
![image](https://github.com/cleong110/sign-language-processing.github.io/assets/122366389/7eeffbca-df03-45c4-992a-9cd2b8024110)

WINNER: SignBERT+ again

**Comment by cleong110** [comment]
OK, here's the transcript of the ChatGPT-assisted rewrite process: https://chatgpt.com/share/f42199e8-bd03-4ec3-9913-e5e832bb2885

My original summary was flawed in a number of ways. 

```
@Zhao2023BESTPretrainingSignLanguageRecognition introduce BEST, a pretraining method based on masked modeling of pose sequences using a coupled tokenization scheme.
The method takes in pose triplet units (left hand, right hand, and upper-body with arms) as inputs.
The pose for each part of the triplet is tokenized into discrete codes [@van_den_Oord_2017NeuralDiscreteRepresentationLearning].
Then masked modeling is employed: any or all of the three parts may be masked, e.g. left hand, or right hand, or body+hand, or all of them...
Unlike @hu2023SignBertPlus, they do not mask multi-frame sequnces ("clips") or sub-frame portions of a pose unit (joints).
They validate their pretraining method isolated ISR (MS-ASL [@dataset:joze2018ms], WLASL [@dataset:li2020word], SLR500 [@huang2019attention3DCNNsSLR] and NMFs-CSL [@hu2021NMFAwareSLR]).
They experiment with both pose-to-gloss and video-to-gloss via fusion with I3D [@carreira2017quo].
Results on these datasets are SOTA compared to previous methods, and quite similar to those of SignBERT+ [@hu2023SignBertPlus]

```


Here's the first rewrite suggestions:
```
@Zhao2023BESTPretrainingSignLanguageRecognition introduced BEST, a pre-training method for sign language recognition based on masked modeling of pose sequences using a coupled tokenization scheme. The method uses pose triplet units, consisting of the left hand, right hand, and upper-body with arms, as inputs. Each part of the pose triplet is tokenized into discrete codes to facilitate the modeling process [@van_den_Oord_2017NeuralDiscreteRepresentationLearning]. Masked modeling is then employed, where any or all of the three parts may be masked, such as the left hand, right hand, or upper-body with hands. Unlike @hu2023SignBertPlus, BEST does not mask multi-frame sequences ("clips") or sub-frame portions of a pose unit (joints). They validated their pre-training method on isolated sign recognition (ISR) datasets: MS-ASL [@dataset
], WLASL [@dataset
], SLR500 [@huang2019attention3DCNNsSLR], and NMFs-CSL [@hu2021NMFAwareSLR]. They experimented with both pose-to-gloss and video-to-gloss tasks, incorporating fusion with I3D [@carreira2017quo]. The results on these datasets demonstrated state-of-the-art (SOTA) performance compared to previous methods and were similar to those achieved by SignBERT+ [@hu2023SignBertPlus].
```

Here's the second:
```
Zhao et al. (2023) introduce BEST, a pretraining framework designed to enhance sign language recognition (SLR) by leveraging BERT's masked modeling approach and a novel coupled tokenization scheme. The method uses pose triplet units, which include the left hand, right hand, and upper-body (including arms), as input data. Each component of the pose triplet unit is tokenized into discrete codes, following the approach by van den Oord et al. (2017). Masked modeling is then applied, where any or all components of the triplet (left hand, right hand, or upper-body) may be masked to learn the hierarchical correlation among them. Unlike SignBERT+ by Hu et al. (2023), the BEST framework does not mask multi-frame sequences ("clips") or sub-frame portions (joints) of a pose unit. The authors validate their pretraining method on isolated sign recognition (ISR) tasks using the MS-ASL [@dataset
], WLASL [@dataset
], SLR500 [@huang2019attention3DCNNsSLR], and NMFs-CSL [@hu2021NMFAwareSLR] datasets. The authors conduct experiments on both pose-to-gloss and video-to-gloss tasks by integrating their method with the Inflated 3D ConvNet (I3D) [@carreira2017quo]. The results demonstrate state-of-the-art (SOTA) performance on all evaluated datasets, showing notable improvements over previous methods and comparable results to SignBERT+ [@hu2023SignBertPlus].
```

**Comment by cleong110** [comment]
In addition the PR had various suggestions, e.g. https://github.com/sign-language-processing/sign-language-processing.github.io/pull/61#discussion_r1634494728

**Comment by cleong110** [comment]
```
@Zhao2023BESTPretrainingSignLanguageRecognition introduce BEST (BERT Pre-training for Sign Language Recognition with Coupling Tokenization), a pre-training method based on masked modeling of pose sequences using a coupled tokenization scheme.
This method takes pose triplet units (left hand, right hand, and upper-body with arms) as inputs, each tokenized into discrete codes [@van_den_Oord_2017NeuralDiscreteRepresentationLearning].
Masked modeling is then applied, where any or all components of the triplet (left hand, right hand, or upper-body) may be masked, to learn hierarchical correlations among them.
Unlike @hu2023SignBertPlus, BEST does not mask multi-frame pose sequences or individual joints. 
The authors validate their pre-training method on isolated sign recognition (ISR) tasks using MS-ASL [@dataset:joze2018ms], WLASL [@dataset:li2020word], SLR500 [@huang2019attention3DCNNsSLR], and NMFs-CSL [@hu2021NMFAwareSLR].
Besides pose-to-gloss, they also experiment with video-to-gloss tasks via fusion with I3D [@carreira2017quo].
Results on these datasets demonstrate state-of-the-art performance compared to previous methods and are comparable to those of SignBERT+ [@hu2023SignBertPlus].
```

**Comment by cleong110** [comment]
Updated: https://chatgpt.com/share/f42199e8-bd03-4ec3-9913-e5e832bb2885

**Comment by cleong110** [comment]
Merged
